pyfecons/costing/calculations/volume.py

```diff
@@ -10,5 +10,16 @@ def calc_volume_ring(height: float, inner: float, outer: float) -> float:
     return Meters3(height * np.pi * (outer ** 2 - inner ** 2))
 
 
-def calc_volume_torus(radius: float, inner: float, outer: float):
-    return Meters3(2 * np.pi * radius * np.pi * (inner + outer) ** 2)
+def calc_volume_outer_hollow_torus(axis_t: Meters, inner: Meters, thickness: Meters) -> Meters3:
+    """
+    Volume of the outer surface of a hollow torus defined by:
+
+    axis_t = major radius
+    thickness = thickness in minor radius of the hollow torus
+    inner = the "inner radius" of the follow torus on the outboard side.
+
+    so the minor radius "a" of the outer part of the hollow torus is:
+
+        a = inner - axis_t + thickness.
+    """
+    return Meters3(2 * np.pi * axis_t * np.pi * (inner - axis_t + thickness)**2)
```

Comment:
@craastad I'm concerned that this calculation is still incorrect. Would it be possible to draw an (ASCII) picture?
* Is this for the volume of a thick shell, like a constant-thickness blanket everywhere around a plasma?
If so I would expect the subtraction of two squared terms.

Is the picture like below?
``` 
              ______
            / _____ \
           / /     \ \
           | |  a  it|
           \ \ ____/ /
            \_______/
```
where here `a = axis_t` is say 4 meters,
`i = inner` is like 1 meter, the distance from the axis to the inside of the shell
`t = thickness` is like 0.5 meters, the thickness of the shell.

Reply:
If so, the volume should be
`(2 \pi a) * (\pi * ((i + t)^2 - i^2))`

Reply:
Or if `inner = R + r = 4 + 1` meters then

` volume = (2 \pi a) * (\pi * ((i - a + t)^2 - (i - a)^2)))`

Reply:
I think there are several confusing things here.  In the documentation, I think:

>     inner = the "inner radius" of the follow torus on the outboard side.
should be changed to
>     inner = the "inner major radius" of the inner part of hollow torus on the outboard side.
to indicate this is a "major radius" (relative to the axis of symmetry) not a "minor radius" (relative to axis_t).

In Chris's comment above, he defines "a = axis_t", but in the doc string, "a" is defined as 
> the minor radius "a" of the outer part of the hollow torus
Also, he defines "i = inner is like 1 meter, the distance from the axis to the inside of the shell" but that is a "minor radius" measurement, while the docstring defines "inner" as a major radius measurement.  

Finally, note that the docstring defines the function to return 
"Volume of the outer surface of a hollow torus"
and *not* 
"Volume of a hollow torus"

I.e., the code calculates the equivalent of:
```
   amin = inner - axis_t + thickness
   vol_full =  2 * np.pi * axis_t * np.pi * amin**2 
```

and not
`   vol_hollow=  2 * np.pi * axis_t * np.pi * (amin**2 - (amin-thickness)**2)`

The latter is what Jacob's formula would calculate, but I think vol_full is what the code wants.  If I recall right, the code later subtracts from vol_full the volume of things inside it to calculate vol_hollow, the volume for this component, as it loops through various components of the "radial build".  

It would probably be clearer to change the structure of the code so that what it needs is vol_hollow and not vol_full.  Then this function could be renamed from

calc_volume_outer_hollow_torus
to
calc_volume_hollow_torus

and it could return vol_hollow from:

   amin = inner - axis_t
  vol_hollow=  2 * np.pi * axis_t * np.pi * ((amin+thickness)**2 - (amin)**2)`

I don't know how such a trivial thing got so complicated, but geometry is hard to discuss in words.

Reply:
Thanks for the comments and discussion on this.

The purpose of this PR is to sync changes from the MIF & IFE ARPAE collabs to the PyFECONs repo.

I'll consider these comments in this PR, but most likely they will be addressed in another PR. It's probably the case that `calc_volume_outer_hollow_torus` is not generalized, but a helper for specific usages of the function. There's been alot of confusion about the tori calculations in the past so it will be good to address them sooner than later. I'll make sure a proper issue is created with these discussion points before merging this PR.